### ISSUE TITLE
doas on Manjaro

### ISSUE THREAD

**Issue by msmafra** [r-darwish/topgrade#863]
## What did you expect to happen?
To use doas as default privileges elevator.

## What actually happened?
It keeps using sudo.
[topgrade-no-doas.txt](https://github.com/r-darwish/topgrade/files/8109825/topgrade-no-doas.txt)

## Additional details
- Which operating system or Linux distribution are you using?
   Manjaro Gnome
- How did you install Topgrade?
   cargo install topgrade cargo-update --force

`topgrade --dry-run`
I am not sure if it is wrong or not. I am not familiar with arch based distros behaviours
 time topgrade --cleanup --yes --dry-run

―― 13:59:51 - System update ――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/yay --pacman pacman -Syu --devel --noconfirm
Dry running: /usr/bin/yay --pacman pacman -Scc --noconfirm

―― 13:59:53 - oh-my-zsh ――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Pulling custom plugins and themes
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-history-substring-search
Would pull /home/marcelo/.oh-my-zsh/custom/themes/powerlevel10k
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Dry running: zsh /home/marcelo/.oh-my-zsh/tools/upgrade.sh

―― 13:59:53 - Gnome Shell extensions ―――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/gdbus call --session --dest org.gnome.Shell.Extensions --object-path /org/gnome/Shell/Extensions --method org.gnome.Shell.Extensions.CheckForUpdates

―― 13:59:54 - Flatpak User Packages ――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/flatpak update --user -y
Dry running: /usr/bin/flatpak uninstall --user --unused

―― 13:59:54 - Flatpak System Packages ――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/flatpak update --system -y
Dry running: /usr/bin/flatpak uninstall --system --unused

―― 13:59:54 - snap ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/doas /usr/bin/snap refresh

―― 13:59:54 - Firmware upgrades ――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/fwupdmgr refresh
Dry running: /usr/bin/fwupdmgr get-updates

―― 13:59:54 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
System update: OK
oh-my-zsh: OK
Gnome Shell Extensions: OK
Flatpak: OK
snap: OK
Firmware upgrades: OK
topgrade --cleanup --yes --dry-run  0,77s user 0,13s system 30% cpu 2,911 total

<details><pre>
―― 11:07:32 - System update ――
Dry running: /usr/bin/yay --pacman pacman -Syu --devel --noconfirm
Dry running: /usr/bin/yay --pacman pacman -Scc --noconfirm
―― 11:07:33 - oh-my-zsh ――
Pulling custom plugins and themes
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Would pull /home/marcelo/.oh-my-zsh/custom/plugins/zsh-history-substring-search
Would pull /home/marcelo/.oh-my-zsh/custom/themes/powerlevel10k
Dry running: zsh /home/marcelo/.oh-my-zsh/tools/upgrade.sh
―― 11:07:34 - Gnome Shell extensions ――
Dry running: /usr/bin/gdbus call --session --dest org.gnome.Shell.Extensions --object-path /org/gnome/Shell/Extensions --method org.gnome.Shell.Extensions.CheckForUpdates
―― 11:07:34 - Flatpak User Packages ――
Dry running: /usr/bin/flatpak update --user -y
Dry running: /usr/bin/flatpak uninstall --user --unused
―― 11:07:34 - Flatpak System Packages ――
Dry running: /usr/bin/flatpak update --system -y
Dry running: /usr/bin/flatpak uninstall --system --unused
―― 11:07:34 - snap ――
Dry running: /usr/bin/doas /usr/bin/snap refresh
―― 11:07:34 - Firmware upgrades ――
Dry running: /usr/bin/fwupdmgr refresh
Dry running: /usr/bin/fwupdmgr get-updates
―― 11:07:34 - Summary ――
System update: OK
config-update: SKIPPED: Cannot find "vim" in PATH
Brew: SKIPPED: Cannot find "brew" in PATH
yadm: SKIPPED: Cannot find "yadm" in PATH
nix: SKIPPED: Cannot find "nix" in PATH
home-manager: SKIPPED: Cannot find "home-manager" in PATH
asdf: SKIPPED: Cannot find "asdf" in PATH
pkgin: SKIPPED: Cannot find "pkgin" in PATH
Git repositories: SKIPPED: No repositories to pull
zr: SKIPPED: Cannot find "zr" in PATH
antibody: SKIPPED: Cannot find "antibody" in PATH
antigen: SKIPPED: Path "/home/marcelo/antigen.zsh" doesn't exist
zplug: SKIPPED: Path "/home/marcelo/.zplug" doesn't exist
zinit: SKIPPED: Path "/home/marcelo/.zinit" doesn't exist
zim: SKIPPED: Path "/home/marcelo/.zim" doesn't exist
oh-my-zsh: OK
fisher: SKIPPED: Cannot find "fish" in PATH
bash-it: SKIPPED: Path "/home/marcelo/.bash_it" doesn't exist
oh-my-fish: SKIPPED: Cannot find "fish" in PATH
fish-plug: SKIPPED: Cannot find "fish" in PATH
tmux: SKIPPED: Path "/home/marcelo/.tmux/plugins/tpm/bin/update_plugins" doesn't exist
TLDR: SKIPPED: Cannot find "tldr" in PATH
pearl: SKIPPED: Cannot find "pearl" in PATH
Gnome Shell Extensions: OK
SDKMAN!: SKIPPED: Path "/home/marcelo/.sdkman/bin/sdkman-init.sh" doesn't exist
apm: SKIPPED: Cannot find "apm" in PATH
fossil: SKIPPED: Cannot find "fossil" in PATH
rustup: SKIPPED: Cannot find "rustup" in PATH
.NET: SKIPPED: Cannot find "dotnet" in PATH
choosenim: SKIPPED: Cannot find "choosenim" in PATH
cargo: SKIPPED: Path "/home/marcelo/.cargo" doesn't exist
Flutter: SKIPPED: Cannot find "flutter" in PATH
Go: SKIPPED: Cannot find "go" in PATH
Emacs: SKIPPED: Cannot find "emacs" in PATH
opam: SKIPPED: Cannot find "opam" in PATH
vcpkg: SKIPPED: Cannot find "vcpkg" in PATH
pipx: SKIPPED: Cannot find "pipx" in PATH
conda: SKIPPED: Cannot find "conda" in PATH
pip3: SKIPPED: Cannot find "pip3" in PATH
stack: SKIPPED: Cannot find "stack" in PATH
tlmgr: SKIPPED: tlmgr must be explicity enabled in the configuration to run in Linux
myrepos: SKIPPED: Cannot find "mr" in PATH
chezmoi: SKIPPED: Cannot find "chezmoi" in PATH
jetpack: SKIPPED: Cannot find "jetpack" in PATH
vim: SKIPPED: Cannot find "vim" in PATH
Neovim: SKIPPED: Cannot find "nvim" in PATH
voom: SKIPPED: Cannot find "voom" in PATH
npm: SKIPPED: NPM root is owned by another user which is not the current user. Set use_sudo = true under the NPM section in your configuration to run NPM as sudo
pnpm: SKIPPED: Cannot find "pnpm" in PATH
deno: SKIPPED: Cannot find "deno" in PATH
composer: SKIPPED: Cannot find "composer" in PATH
krew: SKIPPED: Cannot find "kubectl-krew" in PATH
gem: SKIPPED: Cannot find "gem" in PATH
haxelib: SKIPPED: Cannot find "haxelib" in PATH
sheldon: SKIPPED: Cannot find "sheldon" in PATH
rtcl: SKIPPED: Cannot find "rupdate" in PATH
bin: SKIPPED: Cannot find "bin" in PATH
gcloud: SKIPPED: Cannot find "gcloud" in PATH
micro: SKIPPED: Cannot find "micro" in PATH
raco: SKIPPED: Cannot find "raco" in PATH
spicetify: SKIPPED: Cannot find "spicetify" in PATH
toolbx: SKIPPED: Cannot find "toolbox" in PATH
Flatpak: OK
snap: OK
pacstall: SKIPPED: Cannot find "pacstall" in PATH
pihole: SKIPPED: Cannot find "pihole" in PATH
Firmware upgrades: OK
Restarts: SKIPPED: Cannot find "needrestart" in PATH

</pre></details>

**Comment by r-darwish** [comment]
The screenshot that you showed shows that Topgrade uses doas and not sudo.

**Comment by msmafra** [comment]
I keeps asking for password on Manjaro, which is not my setting for doas.

**Comment by r-darwish** [comment]
I  suppose that it's `yay` that's invoking sudo. See here: https://github.com/Jguer/yay/issues/1194. You need to remove sudo from your system for yay to use it.
